packages/components/tests/integration/components/hds/alert/index-test.js

```diff
@@ -0,0 +1,214 @@
+    await render(
+      hbs`<Hds::Alert @type="inline" @title="yo" @icon={{false}} />`
+    );
+    assert.dom('.flight-icon').doesNotExist();
+  });
+
+  // TEXT (TITLE + DESCRIPTION)
+
+  test('it should render the title when the @title argument is provided', async function (assert) {
+    await render(hbs`<Hds::Alert @type="inline" @title="This is the title" />`);
+    assert.dom(this.element).hasText('This is the title');
+  });
+  test('it should render the description when the @description argument is provided', async function (assert) {
+    await render(
+      hbs`<Hds::Alert @type="inline" @description="This is the description" />`
+    );
+    assert.dom(this.element).hasText('This is the description');
+  });
+  test('it should render both the title and the description when both the @title and @description arguments are provided', async function (assert) {
+    assert.expect(2);
+    await render(
+      hbs`<Hds::Alert @type="inline" @title="This is the title" @description="This is the description" />`
+    );
+    assert.dom('.hds-alert__title').hasText('This is the title');
+    assert.dom('.hds-alert__description').hasText('This is the description');
+  });
+  test('it should render rich HTML when the @description argument contains HTML tags', async function (assert) {
+    assert.expect(8);
+    await render(
+      hbs`<Hds::Alert @type="inline" @description="Hello <strong>strong</strong> and <em>em</em> and <code>code</code> and <a href='#'>link</a>" />`
+    );
+    assert.dom('.hds-alert__description strong').exists().hasText('strong');
+    assert.dom('.hds-alert__description em').exists().hasText('em');
+    assert.dom('.hds-alert__description code').exists().hasText('code');
+    assert.dom('.hds-alert__description a').exists().hasText('link');
+  });
+
+  // ACTIONS
+
+  test('it should render an Hds::Button component yielded to the "actions" container', async function (assert) {
+    assert.expect(2);
+    await render(
+      hbs`<Hds::Alert @type="inline" @description="yo" id="test-alert" as |A|><A.Button @text="I am a button" @size="small" @color="secondary" /></Hds::Alert>`
+    );
+    assert
+      .dom(
+        '#test-alert .hds-alert__actions button.hds-button.hds-button--size-small.hds-button--color-secondary'
+      )
+      .exists()
+      .hasText('I am a button');
+  });
+  test('it should render an Hds::Link::Standalone component yielded to the "actions" container', async function (assert) {
+    assert.expect(2);
+    await render(
+      hbs`<Hds::Alert @type="inline" @description="yo" id="test-alert" as |A|><A.Link::Standalone @icon="plus" @text="I am a link" href="#" @size="small" @color="secondary" /></Hds::Alert>`
+    );
+    assert
+      .dom(
+        '#test-alert .hds-alert__actions a.hds-link-standalone.hds-link-standalone--size-small.hds-link-standalone--color-secondary'
+      )
+      .exists()
+      .hasText('I am a link');
+  });
+
```

Comment:
As I understand a lot of the head scratching on this component was done to make sure it supports conditional statements (or other logic) within the main block. To make this intention more explicit and prevent future refactoring from omitting this, may I suggest we add a test to cover this scenario?

Rough suggestion
```suggestion
  test('it should render an Hds::Button component when conditionally yielded to the "actions" container', async function (assert) {
    await render(hbs`
      <Hds::Alert @type="inline" @title="alert text" id="test-alert" as |A|>
        {{#if true}}
          <A.Button @text="Secondary" @color="secondary" />
        {{/if}}
      </Hds::Alert>
    `);
    assert.dom('#test-alert .hds-alert__actions button.hds-button').exists().hasText('Secondary');
  });
```

Reply:
So, not exactly. What we were testing the other day, was adding a condition(al) inside a named block. The syntax looks like this (pseudo-code):
```
<Hds::Alert @type="inline" @title="alert text" id="test-alert">
    <:actions as |A|>
        {{#if true}}
            <A.Button @text="Secondary" @color="secondary" />
        {{/if}}
    </:actions>
</Hds::Alert>
```
In this case (the one you suggested above) we know it works, it's a normal yield. And even if we wanted to test it, I am not sure we could test what was the problem (the extra visible space above the "actions").

Reply:
I see, it's a problem only with named blocks. Nevermind then 😬